test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1883,6 +1883,49 @@
            (testing "concurrent-depth metric should be 0"
              (is (= 0 (concurrent-depth))))))))))
 
+(deftest block-command-processing
+  (with-test-db
+    (svc-utils/call-with-puppetdb-instance
+      (assoc (svc-utils/create-temp-config)
+        :database *db*
+        :command-processing {:concurrent-writes 1})
+      (fn []
+        (let [dispatcher (get-service svc-utils/*server* :PuppetDBCommandDispatcher)
+              new-producer-ts (now)
+              base-cmd (get-in wire-catalogs [9 :basic])
+              enqueue-catalog (fn []
+                                (enqueue-command
+                                  dispatcher
+                                  (command-names :replace-catalog)
+                                  9
+                                  (:certname base-cmd)
+                                  nil
+                                  (->  base-cmd
+                                       (assoc :producer_timestamp new-producer-ts)
+                                       tqueue/coerce-to-stream)
+                                  ""))
+              cmds-enqueued (CountDownLatch. 1)]
+
+          (with-redefs [cmd/process-message (fn [cmdref
+                                                q command-chan dlo
+                                                delay-pool broadcast-pool write-dbs
+                                                response-chan stats
+                                                blocklist-config maybe-send-cmd-event!
+                                                shutdown-for-ex]
+                                             (.countDown cmds-enqueued))]
```

Comment:
Looks like the indentation may be a bit off inside this `fn`, and if we don't need any of the args, then I might be tempted to just say `(fn [& _] (.countDown cmds-enqueued))`.